I added the CMSIS include paths to tensorflow/lite/micro/tools/make/ext_libs/cmsis.inc
in the context of #41860, assuming they are necessary. Later, I realized that I was using an incompletely patched CMSIS due to #44001. So, after your changes to download_and_extract.sh, the CMSIS include path is not necessary in cmsis.inc (and this is what I understood from @advaitjain and @mansnils). 

If you comment out lines 122...129 in cmsis.inc and then issue the command
`make -j -f tensorflow/lite/micro/tools/make/Makefile TAGS="cmsis-nn" TARGET=cortex_m_gcc_generic CORTEX_M_CORE=M4F microlite`
you will get
`tensorflow/lite/micro/tools/make/downloads/cmsis/CMSIS/DSP/Include/arm_math_types.h:76:10: fatal error: cmsis_compiler.h: No such file or directory`
because
`#include cmsis_compiler.h`
is not patched.

--- a/tensorflow/lite/micro/tools/make/download_and_extract.sh
+++ b/tensorflow/lite/micro/tools/make/download_and_extract.sh
@@ -112,6 +112,7 @@ patch_cmsis() {
   dspfiles+="\|dsp/svm_defines.h"
   dspfiles+="\|dsp/transform_functions.h"
   dspfiles+="\|dsp/utils.h"
+  dspfiles+="\|dsp/arm_helium_utils.h"
   find tensorflow/lite/micro/tools/make/downloads/cmsis \
     \( -name *.c -or -name *.h -or -name *.cpp \) -exec \
     sed -i "s@#include \"\($dspfiles\)\"@#include \"cmsis/CMSIS/DSP/Include/\1\"@g" {} \;
@@ -123,6 +124,11 @@ patch_cmsis() {
   find tensorflow/lite/micro/tools/make/downloads/cmsis \
     \( -name *.c -or -name *.h -or -name *.cpp \) -exec \
     sed -i "s@#include \"\($nnfiles\)\"@#include \"cmsis/CMSIS/NN/Include/\1\"@g" {} \;
+
+  corefiles="cmsis_compiler.h"
+  find tensorflow/lite/micro/tools/make/downloads/cmsis \
+    \( -name *.c -or -name *.h -or -name *.cpp \) -exec \
+    sed -i "s@#include \"\($corefiles\)\"@#include \"cmsis/CMSIS/Core/Include/\1\"@g" {} \;
 
   # Until the fix for https://github.com/ARMmbed/mbed-os/issues/12568 is
   # rolled into Mbed version used on the Arduino IDE, we have to replace

--- a/tensorflow/lite/micro/tools/make/ext_libs/cmsis.inc
+++ b/tensorflow/lite/micro/tools/make/ext_libs/cmsis.inc
@@ -116,7 +116,8 @@ ifneq ($(filter cmsis-nn,$(ALL_TAGS)),)
       $(CMSIS_PATH)CMSIS/DSP/Include/dsp/fast_math_functions.h \
       $(CMSIS_PATH)CMSIS/DSP/Include/dsp/filtering_functions.h \
       $(CMSIS_PATH)CMSIS/DSP/Include/dsp/interpolation_functions.h \
-      $(CMSIS_PATH)CMSIS/DSP/Include/dsp/matrix_functions.h
+      $(CMSIS_PATH)CMSIS/DSP/Include/dsp/matrix_functions.h \
+      $(CMSIS_PATH)CMSIS/Core/Include/cmsis_compiler.h
 
 
     # Need to add the CMSIS Core includes path.

